### PR TITLE
runtime: remove HCM stream error runtime override

### LINKED ISSUE #16007: envoy.reloadable_features.hcm_stream_error_on_invalid_message deprecation
Your change #12498 (http: send GOAWAY for H2 downstreams  for envoy.overload_actions.disable_http_keepalive) introduced a runtime guarded feature. It has been 6 months since the new code has been exercised by default, so it's time to remove the old code path. This issue tracks source code cleanup so we don't forget.

### PR DESCRIPTION
**Pull request by akonradi**
Commit Message: Remove HCM stream error runtime override
Additional Description:
Remove envoy.reloadable_features.hcm_stream_error_on_invalid_message now
that it is old enough.
Risk Level: low
Testing: ran affected tests
Docs Changes: none
Release Notes: document removed override
Platform Specific Features: none
Fixes #16007

### CODE REVIEW

File: test/common/http/utility_test.cc [hunk 1]
```diff
@@ -409,16 +409,6 @@ TEST(HttpUtility, ValidateStreamErrorsWithHcm) {
   EXPECT_TRUE(Envoy::Http2::Utility::initializeAndValidateOptions(http2_options, true, hcm_value)
                   .override_stream_error_on_invalid_http_message()
                   .value());
-
-  {
-    // With runtime flipped, override is ignored.
```
Comment (asraa) [inline test/common/http/utility_test.cc:414]: I think this file no longer needs test runtime header

**Review by asraa — CHANGES_REQUESTED** [review]
Thanks! Small nit
/wait

**Review by asraa — APPROVED** [review]
Thanks! Needs a main merge :(
/wait

**Review by asraa — APPROVED** [review]

**Comment by akonradi** [comment]
It was probably my other cleanup PR, #16041

**Review by asraa — APPROVED** [review]

**Pull request merged**